File: February2021/ContainerWithMostWater.java
class Solution {

    // TC : O(n)
    // SC : O(1)
    public int maxArea(int[] height) {
        int left = 0;
        int right = height.length-1;
        int maxArea = 0;
        while(left<right){
            int currArea = Math.min(height[left], height[right]) *(right - left);
            maxArea  = Math.max(maxArea, currArea);
            if(height[left] < height[right]){
                left++;
            } else{
                right--;
            }
        }
        return maxArea;
    }
}

File: April2022/Kotlin/ContainerwithMostWater.kt
// Author: Shobhit Behl (LC: shobhitbruh)
class Solution {
    fun maxArea(a: IntArray): Int {
        var i:Int=0
        var j:Int=a.size-1
        var ans:Int=0
        while(i<j){
            ans=Math.max(Math.min(a[i],a[j])*(j-i),ans)
            if(a[i]<a[j]){
                i++
            }else if(a[i]>a[j]){
                j--
            }else{
                i++
                j--
            }
        }
        
        return ans
    }
}